tests/unit/C/common/test_config_category.cpp

```diff
@@ -309,6 +309,17 @@ const char* bigCategory =
 			"\"description\": \"Defines\"} " \
 		"}";
 
+const char *optionals = 
+	"{\"item1\" : { "\
+			"\"type\": \"integer\", \"displayName\": \"Item1\", " \
+			"\"value\": \"3\", \"default\": \"3\", " \
+			"\"description\": \"First Item\", " \
+			"\"group\" : \"Group1\", " \
+			"\"rule\" : \"1 = 0\", " \
```

Comment:
`==`

Reply:
Sorry, I don't understand your comment?

Reply:
Example:  IN *optionals* const char, To test that a proposed value is even a rule of the form

`"rule" : "value % 4 == 3"` OR `"rule" : "value  == 3"`

The rule will be executed as part of the validation process in the configuration manager.
Note, if other validation is called for, e.g. minimum or maximum value, then these will be run in addition to the custom rule.

So meaning that rule expression only apply/checks with **value** KV pair. Category won't be created/updated if it does not meet the rule.

Reply:
It really doesn't matter, the test is simply to test if you can retrieve the rule. The code being tested does not look at the content.

Reply:
That's absolutely correct, but it always be better to have clear test data.